fix(circleci): add api route prefix to proxy rewrite path

andrewthauer opened the pull request:
## Hey, I just made a Pull Request!

Looks like this is also a required change for the circleci plugin to work with the recent backend `/api` route changes.

#### :heavy_check_mark: Checklist

<!--- Put an `x` in all the boxes that apply: -->

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

Rugvip reviewed (approved):
Oh wut? Hmm, kinda feeling like the `/api/proxy` prefix should always be rewritten no matter what.

andrewthauer commented:
> Oh wut? Hmm, kinda feeling like the `/api/proxy` prefix should always be rewritten no matter what.

Yeah, maybe the plugin should do that automatically to avoid having to do it in the config?

timja commented on app-config.yaml:
```diff
@@ -18,7 +18,7 @@ proxy:
     changeOrigin: true
     pathRewrite:
-      '^/proxy/circleci/api/': '/'
+      '^/api/proxy/circleci/api/': '/'
```
is this even needed? 

isn't it the default? the Jenkins plugin works without it

andrewthauer replied:
Apparently 🤷. I just confirmed again on master it only works by adding the `/api`.

timja replied:
i meant removing path rewrite, is that what you tried?

andrewthauer replied:
Yeah, your completely right. If I remove the `pathRewrite` and `changeOrigin` works fine. Thanks.

andrewthauer commented:
I've removed the `pathRewrite` as @timja pointed out it's not actually required. Also updated the relevant docs to reflect this change along with the the new `/api/proxy` routing prefix.
